src/EFCore.Cosmos/Metadata/Conventions/CosmosJsonIdConvention.cs

```diff
@@ -0,0 +1,324 @@
+        IConventionEntityTypeBuilder entityTypeBuilder,
+        IConventionEntityType? newBaseType,
+        IConventionEntityType? oldBaseType,
+        IConventionContext<IConventionEntityType> context)
+        => ProcessEntityType(entityTypeBuilder.Metadata, context);
+
+    /// <inheritdoc />
+    public virtual void ProcessEntityTypeAnnotationChanged(
+        IConventionEntityTypeBuilder entityTypeBuilder,
+        string name,
+        IConventionAnnotation? annotation,
+        IConventionAnnotation? oldAnnotation,
+        IConventionContext<IConventionAnnotation> context)
+    {
+        switch (name)
+        {
+            case CosmosAnnotationNames.ContainerName:
+            case CosmosAnnotationNames.PartitionKeyNames:
+                ProcessEntityType(entityTypeBuilder.Metadata, context);
+                break;
+
+            case CosmosAnnotationNames.DiscriminatorInKey:
+                if (oldAnnotation?.Value != null
+                    || !Equals(annotation?.Value, entityTypeBuilder.ModelBuilder.Metadata.GetDiscriminatorInKey()))
+                {
+                    ProcessEntityType(entityTypeBuilder.Metadata, context);
+                }
+                break;
+
+            case CosmosAnnotationNames.AlwaysCreateShadowIdProperty:
+                if (oldAnnotation?.Value != null
+                    || !Equals(annotation?.Value, entityTypeBuilder.ModelBuilder.Metadata.GetAlwaysCreateShadowIdProperty()))
+                {
+                    ProcessEntityType(entityTypeBuilder.Metadata, context);
+                }
+                break;
+        }
+    }
+
+    /// <inheritdoc />
+    public virtual void ProcessForeignKeyOwnershipChanged(IConventionForeignKeyBuilder relationshipBuilder, IConventionContext<bool?> context)
+        => ProcessEntityType(relationshipBuilder.Metadata.DeclaringEntityType, context);
+
+    /// <inheritdoc />
+    public virtual void ProcessKeyAdded(IConventionKeyBuilder keyBuilder, IConventionContext<IConventionKeyBuilder> context)
+        => ProcessEntityType(keyBuilder.Metadata.DeclaringEntityType, context);
+
+    /// <inheritdoc />
+    public virtual void ProcessKeyRemoved(
+        IConventionEntityTypeBuilder entityTypeBuilder,
+        IConventionKey key,
+        IConventionContext<IConventionKey> context)
+        => ProcessEntityType(entityTypeBuilder.Metadata, context);
+
+    /// <inheritdoc />
+    public virtual void ProcessPropertyAdded(IConventionPropertyBuilder propertyBuilder, IConventionContext<IConventionPropertyBuilder> context)
+        => ProcessEntityType(propertyBuilder.Metadata.DeclaringType.ContainingEntityType, context);
+
+    /// <inheritdoc />
+    public virtual void ProcessPropertyRemoved(
+        IConventionTypeBaseBuilder typeBaseBuilder,
+        IConventionProperty property,
+        IConventionContext<IConventionProperty> context)
+        => ProcessEntityType(typeBaseBuilder.Metadata.ContainingEntityType, context);
```

Comment:
Check that the entity type is still in the model.
```suggestion
    {
        if (!typeBaseBuilder.Metadata.IsInModel)
        {
            return;
        }
        
       ProcessEntityType(typeBaseBuilder.Metadata.ContainingEntityType, context);
   }
```

Also, in other `*Removed` methods.